src/rules/typedefWhitespaceRule.ts

```diff
@@ -112,16 +105,138 @@ class TypedefWhitespaceWalker extends Lint.RuleWalker {
+                    scanner.scan() === ts.SyntaxKind.WhitespaceTrivia;
+            }
+
+            const optionValue: string = this.getLeftOption(option);
+            const message = "expected " + ((optionValue === "space") ? "single " : "") + optionValue + " before colon in " + option;
+            this.performFailureCheck(
+                optionValue,
+                hasLeadingWhitespace,
+                hasSeveralLeadingWhitespaces,
+                colonPosition - 1,
+                message
+            );
+        }
+    }
+
+    private checkRight(option: string, node: ts.Node, scanner: ts.Scanner, colonPosition: number): void {
+        let checkTrailingWhitespace: boolean = this.hasRightOption(option);
+
+        if (checkTrailingWhitespace) {
+            let positionToCheck: number = colonPosition + 1 - node.getStart();
+
+            let hasTrailingWhitespace: boolean;
+            if (positionToCheck >= node.getWidth()) {
+                hasTrailingWhitespace = false;
+            } else {
+                scanner.setTextPos(positionToCheck);
+                hasTrailingWhitespace = scanner.scan() === ts.SyntaxKind.WhitespaceTrivia;
+            }
+
+            positionToCheck = colonPosition + 2 - node.getStart();
+
+            let hasSeveralTrailingWhitespaces: boolean;
+            if (positionToCheck >= node.getWidth()) {
+                hasSeveralTrailingWhitespaces = false;
+            } else {
+                scanner.setTextPos(positionToCheck);
+                hasSeveralTrailingWhitespaces = hasTrailingWhitespace &&
+                    scanner.scan() === ts.SyntaxKind.WhitespaceTrivia;
+            }
+
+            const optionValue: string = this.getRightOption(option);
+            const message = "expected " + ((optionValue === "space") ? "single " : "") + optionValue + " after colon in " + option;
+            this.performFailureCheck(
+                optionValue,
+                hasTrailingWhitespace,
+                hasSeveralTrailingWhitespaces,
+                colonPosition + 1,
+                message
+            );
+        }
+    }
+
+    private performFailureCheck(optionValue: string, hasWS: boolean, hasSeveralWS: boolean, failurePos: number, message: string): void {
+        // has several spaces but should have one or none
+        let isFailure: boolean = hasSeveralWS &&
+            (optionValue === "space" || optionValue === "nospace");
+        // has at least one space but should have none
+        isFailure = isFailure || hasWS && optionValue === "nospace";
+        // has at no space but should have at least one
+        isFailure = isFailure || !hasWS &&
+            (optionValue === "space" || optionValue === "spaces");
+
+        if (isFailure) {
+            this.addFailure(this.createFailure(failurePos, 1, message));
+        }
+    }
+
+    private getColonPosition(node: ts.Node): number {
+        const colon: ts.Node = node.getChildren().filter((child: ts.Node) =>
```

Comment:
In general, we leave out type annotations where they can be figured out automatically by tsc, so here, it can just be `child` instead of `child: ts.Node` and `colon` instead of `colon: ts.Node`. (This may apply a few other places in your changes as well)